### ISSUE TITLE
Inverse of "disableStream" should also exist

### ISSUE THREAD

**Issue by Speiger** [kokorin/Jaffree#423]
**Describe the issue**
If you want to filter out only 1 stream source and ignore everything else then you have to forloop over everything else.
Or make a helper function for that specifically. But once a stream is blacklisted you can't get rid of it either.
It would be nice if you had a option to say: "Only these should be allowed" or a "ignore all" and a "remove from Ignore list".

QoL stuff after all.

**Comment by kokorin** [comment]
Could you clarify, if it's possible to disable all streams but 1 using ffmpeg CLI?

**Comment by Speiger** [comment]
@kokorin lets say: My Stream has: Video, Audio, Subtitles, and maybe at random attachments/Data.
I only want: Audio or subtitles for example.
So i have to iterate over everything and explicitly filter out these two. Because 1 there is no array option, 2 there are no disable all and removing of disabling either...

In other words the nice design philosophy you try to provide falls apart because you have to explicit. 

or to simplify the request:
```java
public Input disableAllStreams();
public Input enableStream(StreamType type);
public Input onlyEnableStream(StreamType type);
```

**Comment by kokorin** [comment]
Could you answer my question?

**Comment by Speiger** [comment]
@kokorin yes, how else would you extract audio files from video+audio files or extract subtitles from the three combo.
It is an essential feature.

This is the video specific documentation, it also exists for Subtitles and Audio, Data, Attachment option.
```
-vn (input/output)
As an input option, blocks all video streams of a file from being filtered or being automatically selected or mapped for any output. See -discard option to disable streams individually.

As an output option, disables video recording i.e. automatic selection or mapping of any video stream. For full manual control see the -map option.
```

Here is the stream selection doc:
```
[4 Stream selection](https://ffmpeg.org/ffmpeg.html#toc-Stream-selection-1)
ffmpeg provides the -map option for manual control of stream selection in each output file. Users can skip -map and let ffmpeg perform automatic stream selection as described below. The -vn / -an / -sn / -dn options can be used to skip inclusion of video, audio, subtitle and data streams respectively, whether manually mapped or automatically selected, except for those streams which are outputs of complex filtergraphs.

**Comment by kokorin** [comment]
With ffmpeg CLI it possible to disable stream types on by one, Jaffree allows the same.

It possible to add a method to disable several stream types (not just 1).

Pay attention that ffmpeg has 2 types of video streams: `v` Matches all video streams and `V` Only matches video streams which are not attached pictures. So with naive implementation the method will disable either `v` or `V`, or even both.

Unfortunately ATM I have not much free time to work on this feature request, please feel free to suggest a PR.

**Comment by Speiger** [comment]
@kokorin I know you don't have much time.

Unfortunately it would require me to switch IDEs to fulfill your PR requirements, which i am not willing to do.
Since EditorConfigs are not supported in Eclipse and the plugin that could support it is simply outdated.

And since the other two PRs are in effective limbo, (since i won't fulfill them by blindly guessing what the logs could mean, or you wouldn't allow the closing of them) i am going to refrain from issuing PRs.

Otherwise these would have been PRs in the first place and not Issues.

**Comment by kokorin** [comment]
@Speiger according to this [SO answer](https://stackoverflow.com/a/8417213/4912398) it should be possible to use Checkstyle plugin for Eclipse.
Editorconfig file is there only to help with autoformatting in IDEa, but GH actions don't use it.

**Comment by Speiger** [comment]
![eclipse_qJMgfzKIK9](https://github.com/user-attachments/assets/aa484533-2600-428d-aa97-e6286ac81210)
This was after I applied checkstyle.
Just showing if there are actually changes i missed. Git would have seen it instantly.

Oh yeah also the plugin is deinstalled now too. Don't want to restart my pc randomly.
